### DISCUSSION TITLE
Origin whitelist in supabase

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#6822] [Questions] [answered]
Is there a way to whitelist an origin in supabase for API requests, as public annon and project URL is clearly visible in chrome dev tools under the network tab. so that requests from the whitelisted origin will be only served? 

![image](https://user-images.githubusercontent.com/44422128/168216154-c8c3929c-71c2-4541-826f-8cc524dd7ecc.png)

**Comment by jaitaiwan** [comment]
I would assume that would be based off of the additional callback URLs in your project settings.

**Comment by soedirgo** [comment, marked as answer]
Hey @ijaspreetsinghh,

The `anon` API key is OK to expose as long as your tables' RLS policies are [properly set up](https://supabase.com/docs/guides/auth).

We don't support allowlisting origins. As an alternative, you can use your own proxy server that communicates with the REST API using a `service_role` API key. Then you can configure your proxy server to allowlist origins.

**Comment by leohxj** [comment]
Is there a solution like [Firebase App Check](https://firebase.google.com/docs/app-check)?

**Comment by luketych** [comment]
I am connecting to supabase with tableau. Tableau is pretty dumb and gets its own IP banned several times per day.

How can I fix this if you don't support whitelisting? Or allowlisting or whatever terms you people use these days..

**Comment by luketych** [comment]


**Reply by luketych** [reply]
@soedirgo

**Comment by luketych** [comment]
@soedirgo
